src/node/redeem.rs

```diff
@@ -299,7 +299,9 @@ impl<J: Jet> RedeemNode<J> {
             ) -> Result<Value, Self::Error> {
                 let arrow = data.node.data.arrow();
                 let target_ty = arrow.target.finalize()?;
-                self.bits.read_value(&target_ty).map_err(Error::from)
+                self.bits
+                    .decode_padded_value(&target_ty)
```

Comment:
In 0719071b8f119db5345a62863a8a96ddf8cc594a:

The redeemnode encoding definitely does not use padded witness values. Does this really not break any tests?

Reply:
You are right. The witness values are encoded without padding. It's amazing that our unit tests don't fail.

I will introduce a value decoder with padding (for frames, etc.) and a decoder without padding (for witness data, etc.).

Reply:
The new methods force us to think about the presence / absence of padding, which is good.

Reply:
Thanks! And yes, absolutely. My original code had made the opposite mistake (assuming everything was using unpadded encoding).